core/src/main/java/software/amazon/awssdk/retry/SdkDefaultRetrySettings.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retry;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import software.amazon.awssdk.RetryableException;
+import software.amazon.awssdk.http.HttpStatusCodes;
+
+// TODO: Add some JDK9 sweetness to this class when ready
+public class SdkDefaultRetrySettings {
```

Comment:
Annotate as internal or public? It feels like something we might want to change, so I'd vote for internal...

Reply:
I'm fine with internal in that we can change it but I don't like the full description of internal for this. I'm fine with customers using / depending on this to get at what we are retrying as long as they understand it will change.

Reply:
Internal for now and make it public in the future if we need to? Less surface area to worry about...